src/cli/Cli.ts

```diff
@@ -404,8 +413,12 @@ export class Cli {
       const commandTypes = commandInfo.command.types;
       if (commandTypes) {
         minimistOptions.string = commandTypes.string;
-        minimistOptions.boolean = commandTypes.boolean;
+
+        // Only treat those options as booleans that are included as arg
+        // Otherwise unincluded options will default to false.
+        minimistOptions.boolean = commandTypes.boolean?.filter(optionName => args.some(arg => `--${optionName}` === arg || `-${optionName}` === arg));
```

Comment:
Why is this filtering necessary? Why isn't it sufficient to re-assign the types defined as boolean to minimist?

Reply:
@waldekmastykarz  That's because the comment isn't clear enough :) 

The reason is: if you just assign everything, minimist will parse unused boolean args as `false`, while they should be `undefined` if unused.

Reply:
I changed the comment a bit. Is it better like this? Writing good explanatory comments is hard. 😁

Reply:
> @waldekmastykarz That's because the comment isn't clear enough :)
> 
> The reason is: if you just assign everything, minimist will parse unused boolean args as `false`, while they should be `undefined` if unused.

Got it! So basically a difference between a `bool` and a `bool?` where we need the latter in `set` commands to leave the current value.

Reply:
Correct! We need `bool?` Almost everywhere. If `undefined` we do not include the option value in api calls